### ISSUE TITLE
Default hyperparams for HDP

### ISSUE THREAD

**Issue by zygmuntz** [piskvorky/gensim#399]
I'm looking at the online HDP paper and the earlier online LDA paper (https://www.cs.princeton.edu/~blei/papers/HoffmanBleiBach2010b.pdf), particularly at the interplay of batch size (S), learning rate (kappa) and slow-down parameter (whatever it is), tau.

It seems to me that default settings in gensim (256, 1.0, 64) are inconsistent. First of all, high learning rate produces worse test peplexity and favours smaller batch sizes. For S == 256 kappa == 1.0 is probably to high. Second, from the LDA paper it seems that for S == 256 tau needs to be bigger: they use 1024.

Tau == 64, as in gensim, works for big batches, like 4096. Big batches and low learning rate (0.5) give best results in terms of likelihood.

With the reservation that I don't know how these params affect training time, maybe it would make sense to change the defaults (at least tau for consistency with batch size), for example to

256 / 0.6-0.8 / 1024 or
4096 / 0.5 / 64

**Comment by piskvorky** [comment]
@jesterhazy where do these default values come from?

**Comment by jesterhazy** [comment]
The defaults are (mostly) from http://www.cs.princeton.edu/~chongw/papers/WangPaisleyBlei2011.pdf

Online HDP seems relatively insensitive to batch size as long as it is "large enough" (>= 256), and the total number of batches results in "enough" updates to corpus parameters.

In my own experiments on large corpora of news articles, different values of tau had little impact. Effect of kappa was inconsistent in Wang et al, but 1.0 was significantly better for my data. It is definitely a parameter you should adjust for your own data set.

**Comment by jonaschn** [comment]
I find this resource helpful: https://vb-learning-rate-demo.herokuapp.com to see how the parameters offset (tau) and decay (kappa) affect the learning rate of the online VB method 

The source code by @ecoronado92 could be found [here](https://github.com/ecoronado92/towards_data_science/tree/master/apps/vb_learning_rate).
